### ISSUE TITLE
Tray icon disappeared on Ubuntu 18.04.3 LTS

### ISSUE THREAD

**Issue by ieayoio** [phocean/TopIcons-plus#133]
The main screen can only be on the left or upper side of the sub-screen, otherwise the "wine system tray" tray icon will disappear when it flickers.

OS:Ubuntu 18.04.3 LTS
gnome:3.28.2
software:wechat、qq
installation:sudo apt-get install gnome-shell-extension-top-icons-plus

PS.The latest edition has no effect.

other:https://github.com/wszqkzqk/deepin-wine-ubuntu/issues/135

**Comment by yanyongyu** [comment]
My main monitor is on the right and the wine icons disappear when it flickers too.

Ubuntu 18.04.3

**Comment by ieayoio** [comment]
> My main monitor is on the right and the wine icons disappear when it flickers too.
> 
> Ubuntu 18.04.3

Yes, the main screen must be on the left to work

**Comment by HyperClockUp** [comment]
> The main screen can only be on the left or upper side of the sub-screen, otherwise the "wine system tray" tray icon will disappear when it flickers.
> 
> OS:Ubuntu 18.04.3 LTS
> gnome:3.28.2
> software:wechat、qq
> installation:sudo apt-get install gnome-shell-extension-top-icons-plus
> 
> PS.The latest edition has no effect.
> .
> other:[wszqkzqk/deepin-wine-ubuntu#135](https://github.com/wszqkzqk/deepin-wine-ubuntu/issues/135)

You are right, when i set my main screen top, all icons show and all wine apps works normally

**Comment by phocean** [comment]
Just to make sure that I understood, to reproduce it I should get a Windows app like wechat and launch it with Wine?

And with an external monitor on the left side for instance?

**Comment by ieayoio** [comment]
> Just to make sure that I understood, to reproduce it I should get a Windows app like wechat and launch it with Wine?
> 
> And with an external monitor on the left side for instance?

I think as long as the main screen is on the right or below the sub screen and the wine application tray icon flashes, bugs will appear, not necessarily QQ or wechat..

But if you want to restore my usage scenario, you can do the following:

```
git clone https://github.com/wszqkzqk/deepin-wine-ubuntu.git
```

Switch to the project directory and do the following:

```
./install_2.8.22.sh
```

Download DEB and install:https://mirrors.aliyun.com/deepin/pool/non-free/d/deepin.com.wechat/deepin.com.wechat_2.6.2.31deepin0_i386.deb



But QQ or wechat can only be used if you have an account, which may cause trouble for you, so I think you can replace QQ or wechat with the other wine app with tray icons flashing

**Comment by ieayoio** [comment]
I suspect that as long as the tray icon flickers, not necessarily the wine program, there will be the bug

**Comment by syaoo** [comment]
I tested the native Linux QQ and Wechat (wine), no matter how the display is set, the Linux QQ tray icon will flicker normally, well Wechat has the same problem. I think this bug is caused by Wine.

**Comment by lucas2595** [comment]
Same thing happens to me. If I set my displays as:

![image](https://user-images.githubusercontent.com/24466413/113698056-6fd88680-96aa-11eb-92e1-641a5e8ada04.png)

Flickering doesn't make the icon disappear. But if I set my displays like this:

![image](https://user-images.githubusercontent.com/24466413/113697683-e9bc4000-96a9-11eb-9308-f2ec915af356.png)

Then the flicker makes the icon disappear.

To make the icons flicker, I'm using foobar2000 (a music player). Whenever I interact with the music player (play / pause / next / previous), the icon flickers and disappears.

I've tested not only this extension, but also:
* https://extensions.gnome.org/extension/1503/tray-icons/
* https://extensions.gnome.org/extension/2890/tray-icons-reloaded/
* https://extensions.gnome.org/extension/495/topicons/
* https://extensions.gnome.org/extension/1674/topiconsfix/

and all have the same problem.

**Comment by imyhxy** [comment]
Why does the icon disappear even if I set the main window to the left？

**Comment by ieayoio** [comment]
I've given up using Ubuntu and replaced it with deepin. Deepin comes with a wine called deepin-wine, which is normal

**Comment by imyhxy** [comment]
> I've given up using Ubuntu and replaced it with deepin. Deepin comes with a wine called deepin-wine, which is normal

Using deepin-wine in ubuntu does not get it right.

**Comment by ieayoio** [comment]
Currently, I am using ubuntu22.04, but the issue still exists. However, I have used xubuntu before, and this issue does not exist on xubuntu
